context.go

```diff
@@ -126,6 +126,36 @@ func (c *Ctx) LoadProject(path string) (*Project, error) {
 	return p, nil
 }
 
+// resolveProjectRoot evaluates the root directory and does the following:
+//
+// If the passed path is a symlink outside GOPATH to a directory within a
+// GOPATH, the resolved full real path is returned.
+//
+// If the passed path is a symlink within a GOPATH, we return an error.
+//
+// If the passed path isn't a symlink at all, we just pass through.
+func (c *Ctx) resolveProjectRoot(path string) (string, error) {
+	// Determine if this is a symlink
+	resolved, err := filepath.EvalSymlinks(path)
+	if err != nil {
+		return "", errors.Wrap(err, "resolveProjectRoot")
+	}
+
+	// Not a symlink, move on
+	if resolved == path {
+		return path, nil
+	}
+
+	// Determine if the symlink is within the GOPATH, in which case we're not
+	// sure how to resolve it.
+	if strings.HasPrefix(path, c.GOPATH) {
```

Comment:
Even though it's deprecated (#296), our current best solution here is to use `filepath.HasPrefix()` instead of `strings.HasPrefix()` for better windows compat.